src/routes/safe/components/Apps/index.jsx

```diff
@@ -142,32 +228,50 @@ function Apps({
   // Load apps list
   useEffect(() => {
     const loadApps = async () => {
-      const appsUrl = process.env.REACT_APP_GNOSIS_APPS_URL ? process.env.REACT_APP_GNOSIS_APPS_URL : GNOSIS_APPS_URL
-      const staticAppsList = [`${appsUrl}/compound`, `${appsUrl}/uniswap`]
+      // recover apps from storage:
+      // * third-party apps added by the user
+      // * disabled status for both static and third-party apps
+      const persistedAppList = (await loadFromStorage(APPS_STORAGE_KEY)) || []
+      const list = [...persistedAppList]
+
+      staticAppsList.forEach((staticApp) => {
+        if (!list.some((persistedApp) => persistedApp.url === staticApp.url)) {
+          list.push(staticApp)
+        }
+      })
 
-      const list = [...staticAppsList]
       const apps = []
+      // using the appURL recover app info
       for (let index = 0; index < list.length; index++) {
         try {
-          const appUrl = list[index]
-          const appInfo = await getAppInfoFromUrl(appUrl)
-          const app = { url: appUrl, ...appInfo }
+          const currentApp = list[index]
+
+          const appInfo = await getAppInfoFromUrl(currentApp.url)
+
+          if (appInfo.error) {
+            throw Error()
+          }
 
-          app.id = JSON.stringify({ url: app.url, name: app.name })
-          apps.push(app)
+          appInfo.disabled = currentApp.disabled === undefined ? false : currentApp.disabled
+
+          apps.push(appInfo)
         } catch (error) {
           console.error(error)
         }
       }
 
-      setAppsList([...apps])
+      setAppList([...apps])
+      const firstEnabledApp = apps.find((a) => !a.disabled)
+      if (firstEnabledApp) {
+        onSelectApp(firstEnabledApp.id)
+      }
       setLoading(false)
     }
 
-    if (!appsList.length) {
+    if (!appList.length) {
       loadApps()
     }
-  }, [appsList])
+  }, [])
```

Comment:
for line 282, if no wallet it connected then `network` is `unknown` which is invalid

Reply:
fixed, it's resolved for good with this ticket https://app.zenhub.com/workspaces/safe-multisig-5ce554debb310a35b2f8b6f8/issues/gnosis/safe-react/704